server/src/data-processors/helpers/verify-auth.js

```diff
@@ -0,0 +1,42 @@
+const requiredDependency = require('../../global-helpers/required-dependency');
+const throwError = require('../../global-helpers/throw-error');
+
+function makeVerifyAuth({ jwt = requiredDependency('jwt') } = {}) {
+  const jwtParamHasVerifyMethod = 'verify' in jwt;
+  if (!jwtParamHasVerifyMethod) {
+    requiredDependency('jwt');
+  } 
+  else if (!process.env.AUTHSECRET) {
+    throwError('An "AUTHSECRET" environment variable is required', 'application');
+  } else {
+    return function (token) {
+      let message, tokenBody;
+  
+      if (token && typeof token === 'string') {
+        jwt.verify(token, process.env.AUTHSECRET, (err, decodedTokenBody) => {
+          if (err) {
+            throwError('You need authorisation to access this endpoint.', 'authorisation');
+          } else {
+            message = "Successfully authenticated.";
+            tokenBody = decodedTokenBody;
+          }
+        });
+  
+        return Object.freeze({
+          message,
+          body: tokenBody
+        });
```

Comment:
Coming around to the `Object.freeze()`! The more I think about it, the more useful it seems! Awesome work coming up with it! 🥇